pandas/tests/io/formats/style/test_to_latex.py

```diff
@@ -484,8 +484,131 @@ def test_parse_latex_css_conversion(css, expected):
+@pytest.mark.parametrize(
+    "convert, exp", [(True, "bfseries"), (False, "font-weightbold")]
+)
+def test_parse_latex_css_convert_minimal(styler, environment, convert, exp):
+    # parameters ensure longtable template is also tested
+    styler.highlight_max(props="font-weight:bold;")
+    result = styler.to_latex(convert_css=convert, environment=environment)
+    expected = dedent(
+        f"""\
+        0 & 0 & \\{exp} -0.61 & ab \\\\
+        1 & \\{exp} 1 & -1.22 & \\{exp} cd \\\\
+        \\end{{{environment}}}
+    """
+    )
+    assert expected in result
+
+
 def test_parse_latex_css_conversion_option():
     css = [("command", "option--latex--wrap")]
     expected = [("command", "option--wrap")]
     result = _parse_latex_css_conversion(css)
     assert result == expected
+
+
+def test_longtable_comprehensive(styler):
+    result = styler.to_latex(
+        environment="longtable", hrules=True, label="fig:A", caption=("full", "short")
+    )
+    expected = dedent(
+        """\
+        \\begin{longtable}{lrrl}
+        \\caption[short]{full} \\label{fig:A} \\\\
+        \\toprule
+        {} & {A} & {B} & {C} \\\\
+        \\midrule
+        \\endfirsthead
+        \\caption[]{full} \\\\
+        \\toprule
+        {} & {A} & {B} & {C} \\\\
+        \\midrule
+        \\endhead
+        \\midrule
+        \\multicolumn{4}{r}{Continued on next page} \\\\
+        \\midrule
+        \\endfoot
+        \\bottomrule
+        \\endlastfoot
+        0 & 0 & -0.61 & ab \\\\
+        1 & 1 & -1.22 & cd \\\\
+        \\end{longtable}
+    """
+    )
+    assert result == expected
+
+
+def test_longtable_minimal(styler):
+    result = styler.to_latex(environment="longtable")
+    expected = dedent(
+        """\
+        \\begin{longtable}{lrrl}
+        {} & {A} & {B} & {C} \\\\
+        \\endfirsthead
+        {} & {A} & {B} & {C} \\\\
+        \\endhead
+        \\multicolumn{4}{r}{Continued on next page} \\\\
+        \\endfoot
+        \\endlastfoot
+        0 & 0 & -0.61 & ab \\\\
+        1 & 1 & -1.22 & cd \\\\
+        \\end{longtable}
+    """
+    )
+    assert result == expected
+
+
+@pytest.mark.parametrize(
+    "sparse, exp",
+    [(True, "{} & \\multicolumn{2}{r}{A} & {B}"), (False, "{} & {A} & {A} & {B}")],
+)
+def test_longtable_multiindex_columns(df, sparse, exp):
+    cidx = MultiIndex.from_tuples([("A", "a"), ("A", "b"), ("B", "c")])
+    df.columns = cidx
+    expected = dedent(
+        f"""\
+        \\begin{{longtable}}{{lrrl}}
+        {exp} \\\\
+        {{}} & {{a}} & {{b}} & {{c}} \\\\
+        \\endfirsthead
+        {exp} \\\\
+        {{}} & {{a}} & {{b}} & {{c}} \\\\
+        \\endhead
+        """
+    )
+    assert expected in df.style.to_latex(environment="longtable", sparse_columns=sparse)
+
+
+@pytest.mark.parametrize("caption", ["full", ("full", "short")])
+@pytest.mark.parametrize("label", [None, "tab:A"])
+def test_longtable_caption_label(styler, caption, label):
+    if isinstance(caption, str):
```

Comment:
I suggest against conditional logic in tests.
If parametrization makes this test so complex, then better not to parametrize.

Reply:
I do in general agree with your philosophy, although in this case as you observed the logic is an extension to the parametrization, and the test is isomorphic to one with:
```
@pytest.mark.parametrize("caption, cap1, cap2", [(...), (...)])
@pytest.mark.parametrize("label, lab", [(...), (...])
def test_longtable_caption_label(styler, caption, cap1, cap2, label, lab): 
# no conditional logic.
```
I don't see this as a blocker, maybe just artistic difference! I also take your point about complexity but think this test is still pretty low level in that too.

Reply:
You could put the expected macros right here to avoid any logic and make tests more readable:
```
@pytest.mark.parametrize(
    "caption, caption_macro",
    [
        ("full", "\caption{full}"),
        (("full", "short"), "\caption[short]{full}"),
    ],
)
@pytest.mark.parametrize(
    "label, label_macro",
    [
        (None, ""),
        ("tab:A", "\label{tab:A}"),
    ],
)
```

This way, will not need branching inside the test.

Reply:
But in your case there will be cap1 and cap2, rather than a single caption macro.

Reply:
I believe this is more readable the way I previously structured it. Never-the-less I have changed it to you preference which is also fine with me.